contracts/contracts/RequestManager.sol

```diff
@@ -371,6 +371,31 @@ contract RequestManager is Ownable, LpWhitelist, RestrictedCalls, Pausable {
         validRequestId(requestId)
         onlyAllowed(msg.sender)
         returns (uint96)
+    {
+        return claimRequest(msg.sender, requestId, fillId);
+    }
+
+    /// Claim that a request was filled by a claimer.
+    ///
+    /// The request must still be valid at call time.
+    /// The caller must provide the ``claimStake`` amount of source rollup's native
+    /// token.
+    /// Only the claimer may get the stake back later.
+    ///
+    /// @param claimer Address of the claimer.
+    /// @param requestId ID of the request.
+    /// @param fillId The fill ID.
+    /// @return The claim ID.
+    function claimRequest(
+        address claimer,
+        bytes32 requestId,
+        bytes32 fillId
+    )
```

Comment:
While this is totally fine, I think we should really reconsider the naming here.
For example,
```
    /// Claim that a request was filled by a claimer.
```
looks confusing and while I understand what is being meant here, we're not making it easy for other people.

I could not come up with an ideal naming, but my thinking goes like this:
- the word `claimer` should be used to denote whoever actually called `claimRequest`, regardless of the variant of that function
- the word `filler` should be used to denote whoever the `claimer` claims to have been the filler of the request

With this we would not need to distinguish between a "claimer" and an "initiator".
With such naming, we would also replace the `claimer` field in `Claim` struct with a `filler` field:
```
struct Claim {                                    
    bytes32 requestId;                            
-    address claimer;                              
+   address filler;                              
    uint256 claimerStake;                         
    ...
}                                                 
```

This may look a slightly confusing at first, but it does make sense: the `claimer` claims that `filler` filled the request. We may not know whether that is true or not yet, until the challenge game is done. Also note that `claimerStake` is unchanged -- it is the `claimer` that puts (and possibly loses) their stake, not the `filler`.

So I really think we should consider changing naming here to reflect the new possibilities, but at the same time, if you decide to do it, I would suggest to do it in another PR, to keep this one simple.

Reply:
I get your point, although I'm not really happy to use the term filler as long as it is unclear whether it is really the filler. For example, to my understanding, a filler is the real filler of the request. In the Request Manager any address is accepted as the filler once a) a claim is finalized and valid or b) l1 resolved. In both cases we write the `filler` in the request struct.
It would be confusing to me to have a bunch of claims with lots of different fillers. So I would really consider keeping a unique name for a valid and accepted filler, and for an address who is claiming to be the filler.
Additionally, we also need a good naming for whoever called the function.

> Also note that claimerStake is unchanged -- it is the claimer that puts (and possibly loses) their stake, not the filler.

This is not quite true, or at least I would not argue in the way you just did. Technically, the msg.sender puts in the claim stake at first, but it is denominated to be the stake of the (in your case) filler. To be more precise, allthough the claimer puts in the first claim stake, any other counter challenge the actual filler has to increase its stake. At the end all of the stake goes back to the filler.
Note, that how the filler and the claimer deal with this "transfer" of claim stake from the claimer to the filler is out of the protocol and is not "our business". 

In any case, I would not name the stake after the caller of the transaction.